app/proguard-rules.pro

```diff
@@ -107,4 +107,15 @@
     java.lang.String getId();
     boolean isLimitAdTrackingEnabled();
 }
--keep public class com.android.installreferrer.** { *; }
\ No newline at end of file
+-keep public class com.android.installreferrer.** { *; }
+
+# Keep Android Lifecycle methods
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1596302
+# https://twitter.com/rakshakhegde/status/875107392198856706
+-keep public class android.arch.lifecycle.** {
```

Comment:
Sanity check: Is this correct even though Glean imports it as `androidx.lifecycle`?

https://github.com/mozilla/glean/blob/master/glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/GleanLifecycleObserver.kt#L7

Reply:
Ah yes, good catch